### PR TITLE
[cloudian] Allow storage class to define IAM policy document.

### PR DESCRIPTION
**Pull request by nberrington-cloudian**
1) If stroage class has policy document, use that instead of the default read+write policy
2) Pretty print stored policy document to make it easier to review from CMC
3) Update DEPLOY documentation and storageclass.yaml examples.

### CODE REVIEW

File: cmd/iam.go [hunk 1]
```diff
@@ -215,27 +216,38 @@ func (p *awsS3Provisioner) createBucketPolicyDocument(bktName string) (string, e
...
+				default:
+					return "", fmt.Errorf("unknown permission, %s", *spec.LocalPermission)
+				}
+			}
+		*/
+		policy.Statement = append(policy.Statement, read, write)
+	}
 
-	b, err := json.Marshal(&policy)
+	b, err := json.MarshalIndent(&policy, "", "  ")
```
Comment (sjohnson-cloudian) [inline cmd/iam.go:250]: Does this change have any material effect?

File: examples/greenfield/storageclass.yaml [hunk 2]
```diff
@@ -12,7 +12,23 @@ parameters:
...
+  #iamPolicy: |
+  #  {
+  #    "Version": "2012-10-17",
+  #    "Statement": [{
+  #            "Sid": "AllowAll",
+  #            "Effect": "Allow",
+  #            "Action": ["s3:HeadObject", "s3:ListBucket", "s3:GetObject"]
+  #    }]
+  #  }
+
```
Comment (sjohnson-cloudian) [inline examples/greenfield/storageclass.yaml]: This example is counter-intuitive for greenfield!  You'll just get an empty bucket that you can see is empty and no more!

File: cmd/iam.go (hunk 1 above)
Comment (nberrington-cloudian) [reply cmd/iam.go:250]: No, but it makes it easier to read the policy from the CMC

File: examples/greenfield/storageclass.yaml (hunk 2 above)
Comment (nberrington-cloudian) [reply examples/greenfield/storageclass.yaml]: I'll change it to write only

Comment (sjohnson-cloudian) [reply examples/greenfield/storageclass.yaml]: Does that make sense either?  A greenfield bucket is not accessibly by anyone else...

Comment (nberrington-cloudian) [reply examples/greenfield/storageclass.yaml]: Ok, I'll remove it from greenfield entirely then :-p

Comment (sjohnson-cloudian) [reply examples/greenfield/storageclass.yaml]: Makes sense - am struggling to see why a greenfield bucket wouldn't just have full access.

**Comment by sjohnson-cloudian** [comment]
LGTM.

**Pull request merged**